File: SDKs/amazonaws.com/waf/2015-08-24/python/src/sdk/models/shared/createratebasedrulerequest.py
from dataclasses import dataclass, field
from typing import List,Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class CreateRateBasedRuleRequest:
    change_token: str = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('ChangeToken') }})
    metric_name: str = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('MetricName') }})
    name: str = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('Name') }})
    rate_key: RateKeyEnum = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('RateKey') }})
    rate_limit: int = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('RateLimit') }})
    tags: Optional[List[Tag]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Tags') }})
    

File: SDKs/amazonaws.com/waf/2015-08-24/typescript/src/sdk/models/shared/createratebasedrulerequest.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { RateKeyEnum } from "./ratekeyenum";
import { Tag } from "./tag";



export class CreateRateBasedRuleRequest extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=ChangeToken" })
  changeToken: string;

  @SpeakeasyMetadata({ data: "json, name=MetricName" })
  metricName: string;

  @SpeakeasyMetadata({ data: "json, name=Name" })
  name: string;

  @SpeakeasyMetadata({ data: "json, name=RateKey" })
  rateKey: RateKeyEnum;

  @SpeakeasyMetadata({ data: "json, name=RateLimit" })
  rateLimit: number;

  @SpeakeasyMetadata({ data: "json, name=Tags", elemType: Tag })
  tags?: Tag[];
}
